uniprot-search/src/test/java/org/uniprot/store/search/SolrQueryUtilTest.java

```diff
@@ -211,4 +216,25 @@ void hasNegativeTermComplexQueryReturnsFalse() {
         boolean result = SolrQueryUtil.hasNegativeTerm(inputQuery);
         assertFalse(result);
     }
+
+    @ParameterizedTest
+    @MethodSource("getQueryWithExpectedResult")
+    void hasLeadingWildcardTermTrue(String query, boolean expectedResult) {
+        boolean actualResult =
+                SolrQueryUtil.ignoreLeadingWildcard(query, Set.of("gene", "protein_name"));
+        assertEquals(expectedResult, actualResult, query + " failed");
+    }
+
+    private static Stream<Arguments> getQueryWithExpectedResult() {
+        return Stream.of(
+                Arguments.of("*quick brown fox", true),
+                Arguments.of("quick brown fox*", false),
+                Arguments.of("quick * brown fox", false),
+                Arguments.of("qui*ck", false),
+                Arguments.of("otherfield:*quick brown fox", true),
+                Arguments.of("*brown fox AND otherfield:quick", true),
+                Arguments.of("brown fox AND otherfield:quick", false),
+                Arguments.of("gene:*brown", false),
+                Arguments.of("protein_name:*brown AND reviewed:*true", true));
```

Comment:
I just realised another use case, Add one query with phrasal query (query with double quote) for example:Arguments.of("\"*quick brow\"", false)